packages/react/src/components/header/Header.tsx

```diff
@@ -19,21 +27,52 @@ export type HeaderProps = React.PropsWithChildren<{
    * ID of the header element.
    */
   id?: string;
-}>;
+}
+
+export interface HeaderProps extends HeaderNodeProps {
+  onDidChangeLanguage?: (string) => void;
+}
 
-export const Header = ({ children, className, id }: HeaderProps) => {
-  const isSmallScreen = useMediaQueryLessThan('m');
-  const isMediumScreen = useMediaQueryLessThan('l');
-  const context: HeaderContextProps = { isSmallScreen, isMediumScreen };
+const HeaderNode: ComponentType<HeaderNodeProps> = ({ children, className, ...props }) => {
+  const { isNotLargeScreen } = useHeaderContext();
+  const headerClassNames = classNames('hds-header', styles.header, className, { isNotLargeScreen });
   return (
-    <HeaderContext.Provider value={context}>
-      <header id={id} className={classNames(styles.header, className)}>
-        <div className={styles.headerBackgroundWrapper}>{children}</div>
-      </header>
-    </HeaderContext.Provider>
+    <header className={headerClassNames} {...props}>
+      <div className={styles.headerBackgroundWrapper}>{children}</div>
+    </header>
+  );
+};
+
+interface HeaderInterface extends FC<HeaderProps> {
+  UniversalBar: typeof HeaderUniversalBar;
+  ActionBar: typeof HeaderActionBar;
+  NavigationMenu: typeof HeaderNavigationMenu;
+  ActionBarItem: typeof HeaderActionBarItemWithDropdown;
+  NavigationLink: typeof NavigationLink;
+  NavigationLanguageSelector: typeof NavigationLanguageSelector;
+  NavigationSearch: typeof NavigationSearch;
+}
+
+export const Header: HeaderInterface = ({
+  onDidChangeLanguage,
+  defaultLanguage,
+  ...props
+}: HeaderProps & LanguageProviderProps) => {
+  return (
+    <HeaderContextProvider>
+      <LanguageProvider onDidChangeLanguage={onDidChangeLanguage} defaultLanguage={defaultLanguage}>
```

Comment:
I was confused at first that do we really need LanguageProvider, but yeah we do because the language selector works funkily as it might move a language selected from the menu to the top to show as selected. Good job with this 👍